auth-tokens/src/test/java/com/palantir/tokens/auth/UnverifiedJsonWebTokenTests.java

```diff
@@ -58,17 +58,25 @@ public void testAsJwt_validJwtFromSessionToken() {
     @Test
     public void testAsJwt_validJwtFromApiToken() {
         UnverifiedJsonWebToken token = UnverifiedJsonWebToken.of(API_TOKEN);
-        assertEquals(USERID, token.getUnverifiedUserId());
-        assertEquals(Optional.empty(), token.getUnverifiedSessionId());
-        assertEquals(Optional.of(TOKEN_ID), token.getUnverifiedTokenId());
+        assertValidToken(token);
     }
 
     @Test
     public void testAsJwt_validJwtFromProxyToken() {
         UnverifiedJsonWebToken token = UnverifiedJsonWebToken.of(PROXY_TOKEN);
-        assertEquals(USERID, token.getUnverifiedUserId());
-        assertEquals(Optional.empty(), token.getUnverifiedSessionId());
-        assertEquals(Optional.of(TOKEN_ID), token.getUnverifiedTokenId());
+        assertValidToken(token);
+    }
+
+    @Test
+    public void testAsJwt_validJwtFromParsedToken() {
+        Optional<UnverifiedJsonWebToken> token = UnverifiedJsonWebToken.tryParse(PROXY_TOKEN.getToken());
+        assertValidToken(token.get());
+    }
+
+    @Test
+    public void invalidJwt_parseReturnsEmpty() {
+        Optional<UnverifiedJsonWebToken> parsedJwt = UnverifiedJsonWebToken.tryParse(INVALID_BEARER_TOKEN.getToken());
```

Comment:
worth a test for the case where the token has two periods in, but still isn't a valid bearer token